sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java

```diff
@@ -1377,25 +1386,38 @@ public void startBundle(StartBundleContext c) throws IOException {
       }
 
       badRecords = new HashSet<>();
+      outstandingWrites = new LinkedBlockingDeque<>();
     }
 
     @ProcessElement
     public void processElement(ProcessContext c, BoundedWindow window) throws Exception {
+      // burn down the completed futures to avoid unbounded memory growth
+      for (Future<?> f = outstandingWrites.peek();
```

Comment:
I don't think we want this, doesn't this essentially remove batching?

Reply:
Not sure what you mean? this shouldnt block anything or affect batching

Reply:
oh never mind yeah sorry I misread this, can you refactor it out into another method though?

Reply:
done